### ISSUE TITLE
NoSuchMethodError: The method 'cancel' was called on null.

### ISSUE THREAD

**Issue by Mike278** [ReactiveX/rxdart#478]
Discovered this while reproducing #477 so possibly related - not super familiar with the codebase. These tests pass in 23.1 and fail in 24.1.

```dart
test("v1", () async {
  final a = BehaviorSubject.seeded('a');
  final b = BehaviorSubject.seeded('b');
  final bug = Rx.combineLatest2(a, b, (_a, _b) => 'ab').shareValue();
  expect(await bug.first, 'ab');
  expect(await bug.first, 'ab');  // NoSuchMethodError
});
```
```
dart:async                                             _BroadcastStreamController.add
package:rxdart/src/transformers/start_with.dart 56:12  _StartWithStreamSink._safeAddFirstEvent
package:rxdart/src/transformers/start_with.dart 37:11  _StartWithStreamSink.onListen
package:rxdart/src/utils/forwarding_stream.dart 31:37  forwardStream.<fn>.<fn>
package:rxdart/src/utils/forwarding_stream.dart 24:12  forwardStream.runCatching
package:rxdart/src/utils/forwarding_stream.dart 31:16  forwardStream.<fn>
dart:async                                             _StreamImpl.listen
package:rxdart/src/streams/defer.dart 37:18            DeferStream.listen
dart:async                                             Stream.first
test\rxdart_repro_test.dart 15:21                   main.<fn>
===== asynchronous gap ===========================
dart:async                                             _StreamImpl.listen
package:rxdart/src/streams/defer.dart 37:18            DeferStream.listen
dart:async                                             Stream.first
test\rxdart_repro_test.dart 15:21                   main.<fn>
===== asynchronous gap ===========================
dart:async                                             _asyncThenWrapperHelper
test\rxdart_repro_test.dart                         main.<fn>

NoSuchMethodError: The method 'cancel' was called on null.
Receiver: null
Tried calling: cancel()
```

```dart
test("v2", () async {
  final a = BehaviorSubject.seeded('a', sync: true);
  final bug = a.switchMap((_) => BehaviorSubject.seeded('b', sync: true));
  expect(await bug.first, 'b'); // NoSuchMethodError
});
```
```
dart:async                                             _BroadcastStreamController.add
package:rxdart/src/transformers/start_with.dart 56:12  _StartWithStreamSink._safeAddFirstEvent
package:rxdart/src/transformers/start_with.dart 37:11  _StartWithStreamSink.onListen
package:rxdart/src/utils/forwarding_stream.dart 31:37  forwardStream.<fn>.<fn>
package:rxdart/src/utils/forwarding_stream.dart 24:12  forwardStream.runCatching
package:rxdart/src/utils/forwarding_stream.dart 31:16  forwardStream.<fn>
dart:async                                             _StreamImpl.listen
package:rxdart/src/streams/defer.dart 37:18            DeferStream.listen
dart:async                                             StreamView.listen
package:rxdart/src/transformers/switch_map.dart 19:40  _SwitchMapStreamSink.add
package:rxdart/src/utils/forwarding_stream.dart 34:49  forwardStream.<fn>.<fn>.<fn>
package:rxdart/src/utils/forwarding_stream.dart 24:12  forwardStream.runCatching
package:rxdart/src/utils/forwarding_stream.dart 34:28  forwardStream.<fn>.<fn>
dart:async                                             _BroadcastStreamController.add
package:rxdart/src/transformers/start_with.dart 56:12  _StartWithStreamSink._safeAddFirstEvent
package:rxdart/src/transformers/start_with.dart 37:11  _StartWithStreamSink.onListen
package:rxdart/src/utils/forwarding_stream.dart 31:37  forwardStream.<fn>.<fn>
package:rxdart/src/utils/forwarding_stream.dart 24:12  forwardStream.runCatching
package:rxdart/src/utils/forwarding_stream.dart 31:16  forwardStream.<fn>
dart:async                                             _StreamImpl.listen
package:rxdart/src/streams/defer.dart 37:18            DeferStream.listen
dart:async                                             StreamView.listen
package:rxdart/src/utils/forwarding_stream.dart 33:27  forwardStream.<fn>
dart:async                                             Stream.first
test\rxdart_repro_test.dart 21:20                   main.<fn>
===== asynchronous gap ===========================
dart:async                                             Stream.first
test\rxdart_repro_test.dart 21:20                   main.<fn>

NoSuchMethodError: The method 'cancel' was called on null.
Receiver: null
Tried calling: cancel()
```

The last example does not crash if you change to `sync: false` 
```dart
test("v3", () async {
  final a = BehaviorSubject.seeded('a', sync: false);
  final bug = a.switchMap((_) => BehaviorSubject.seeded('b', sync: false));
  expect(await bug.first, 'b');
  expect(await bug.first, 'b'); 
  // passes
});
```

**Comment by hoc081098** [comment]
Related https://github.com/dart-lang/sdk/issues/41969 and  https://github.com/ReactiveX/rxdart/issues/463

**Comment by hoc081098** [comment]
`shareValue` operartor:
RxDart 23.1: adds event asynchronously by default (`scheduleMicrotask`)
RxDart 24.1: adds event synchronously

**Comment by frankpepermans** [comment]
If sync, then startWith immediately adds the even when onListen triggers, 
At this point, in the sdk, the underlying subscription is not yet ready,
In this case, it then wants to cancel it right after the first value is received, causing a NPE.

See https://github.com/dart-lang/sdk/issues/41969

So, ideally we do not add on onListen, problem is, there isn't really a different way of pushing that first event immediately, before any other event will be added...

I think we can only fix this by adjusting our Subjects, so that get value (behavior) and values (replay) are filled immediately, but the event(s) not.

**Comment by frankpepermans** [comment]
should be fixed with 0.25.0-beta

**Comment by jonsamwell** [comment]
0.25.0-beta fixes it for me when using `shareValue`
